### PR TITLE
chore: dashboard provider use context selector

### PR DESCRIPTION
**Pull request by IrakliJani**
### Description:

- adding `use-context-selector` to dashboard context
- we use this library for explore contexts → see `useExplorerContext`

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/providers/DashboardProvider.tsx [hunk 1]
```diff
@@ -493,12 +492,15 @@ export const DashboardProvider: React.FC = ({ children }) => {
...
-    }
-    return context;
-};
+export function useDashboardContext<Selected>(
+    selector: (value: DashboardContext) => Selected,
+) {
+    return useContextSelector(Context, (context) => {
+        if (context === undefined) {
+            throw new Error(
+                'useExplorer must be used within a ExplorerProvider',
```
Comment (joaoviana) [inline packages/frontend/src/providers/DashboardProvider.tsx:501]:
```suggestion
                'useDashboardContext must be used within a DashboardProvider',
```

**Review by joaoviana — CHANGES_REQUESTED** [review]
Checking in Render once available 👍

File: packages/frontend/src/providers/DashboardProvider.tsx (hunk 1 above)
Comment (IrakliJani) [reply packages/frontend/src/providers/DashboardProvider.tsx]: nice catch ! caught me on copycatting :D

**Comment by IrakliJani** [comment]
@joaoviana updated PR and tests are passing now

**Review by joaoviana — APPROVED** [review]
Tested a few core journeys on creating and viewing a dashboard 👍

**Pull request merged**